Lib/test/test_reprlib.py

```diff
@@ -580,6 +580,42 @@ def test_invalid_indent(self):
                 with self.assertRaisesRegex(expected_error, expected_msg):
                     r.repr(test_object)
 
+    def test_shadowed_builtin(self):
+        # Issue #113570: repr() should not be fooled by a shadowed builtin
+        # function
+        class array:
```

Comment:
`array` is not really a builtin :)
It is a stdlib member. Please, also add a test for the builtin module.